src/net/events.ts

```diff
@@ -560,3 +561,48 @@ export function addEventComponents(em: EntityManager) {
   em.addSingletonComponent(DetectedEventsDef);
   em.addSingletonComponent(EventsDef);
 }
+
+export function eventWizard<ES extends EDef<any>[], Extra>(
+  name: string,
+  entities: readonly [...ES] | (() => readonly [...ES]),
+  runEvent: (entities: ESet<ES>, extra: Extra) => void,
+  opts?: {
+    legalEvent?: (entities: ESet<ES>, extra: Extra) => boolean;
+    eventAuthorityEntity?: (entityIds: NumberTuple<ES>) => number;
+  } & Partial<ExtraSerializers<Extra>>
+): (es: ESet<ES>, extra?: Extra) => void {
+  const delayInit = typeof entities === "function";
+  const initThunk = () => {
+    registerEventHandler<ES, Extra>(name, {
+      entities: delayInit ? entities() : entities,
+      eventAuthorityEntity: (es) => {
+        if (opts?.eventAuthorityEntity) return opts.eventAuthorityEntity(es);
+        else return es[0];
+      },
+      legalEvent: (em, es, extra) => {
+        if (opts?.legalEvent) return opts.legalEvent(es, extra);
+        return true;
+      },
+      runEvent: (em, es, extra) => {
+        runEvent(es, extra);
+      },
+      ...(opts?.serializeExtra ? { serializeExtra: opts.serializeExtra } : {}),
+      ...(opts?.deserializeExtra
+        ? { deserializeExtra: opts.deserializeExtra }
+        : {}),
+    });
+  };
+
+  if (delayInit) onInit(initThunk);
+  else initThunk();
+
+  const raiseEvent = (es: ESet<ES>, extra?: Extra) => {
```

Comment:
Could we make this take either objects or IDs? We may sometimes want to raise an event w/o having the entities already (e.g., the `hat-drop` event  in `hat.ts`)

Reply:
oh hmm should be do-able, let me try